### PR TITLE
4/7 Changes

### PR DESCRIPTION
**Pull request by beansrowning**
# New workflow  

From clone, you'll just run two scripts to pre-process data and run all models for every outcome:  

1. run_preprocessing.sh
2. run_all_models.sh  

# Changes

## Shell Scripts  
- run_all_models.sh is the new script for running all models across all outcomes and producing CIs
    - This could still use some error handling to halt the script on failure, but generally works.   
- run_models.sh now doesn't run CI script by default  
- run_models.sh also runs class-weighted fit for DAN and LSTM by default

## CI Script (python/analysis.py)
- CI script now checks the predictions spreadsheet for the model names, so there's no need for these to be pre-specified  
- new `--parallel` flag switches BCa computation to parallel execution (defaults false, but this is specified in run_all_models)  

## Baseline Models (python/baseline_models.py)  
- Out and data directories are now optional and do not have a default value
- All paths are constructed relative to script path, with command line arguments available to override if needed
- Subdirectories of the output folder are created if they don't exist at runtime to avoid errors when we go to write out stats
- All probs dumped into pickle dict with cutpoint used for downstream analysis

## Keras Models DAN/LSTM (python/model.py)
- Default batch size increased to 128, which apparently works now and fits much quicker
- Paths handled in same way as baseline models to avoid path construction issues
- weighted loss added as command line option to use class weights in loss computation
- TARGET -> OUTCOME to align with variable name in other scripts
- tf.data generator shennanigans resolved by setting epochs = 1, which doesn't really make a lot of sense but ¯\\_(ツ)_/¯  
- DAN and LSTM stats write-out streamlined

## Stats helpers (python/tools/analysis.py)  
- Whitespace changes, PEP8 compliance via yapf
- ROC-AUC error in multiclass case resolved by softmaxing logits from the sigmoid output of LSTM/DAN  
- Still 2 merge_cis() functions (@scotthlee which one do we want?)  
- imports streamlined based on usage

## Multiprocessing helpers (python/tools/multi.py)  
- boot_cis and jacknife_metrics now mirror non-Pool variants (but this should probably be handled as one entity to avoid divergence)  
- Streamlined imports based on usage

### CODE REVIEW

File: python/analysis.py [hunk 1]
```diff
@@ -13,19 +15,24 @@
...
-                        dest='day_one',
-                        action='store_false')
+    parser.add_argument(
+        '--all_days',
+        help=
+        "Only compute CIs for models that used all features in lookback period",
+        dest='day_one',
+        action='store_false')
     parser.set_defaults(day_one=True)
     parser.add_argument("--outcome",
```
Comment (beansrowning) [inline python/analysis.py:29]: @scotthlee I worked in some of your changes to the analysis script.

Basically it should do the same thing you wanted by looping over each outcome, but it also allows you to specify one or more outcomes you want specifically through a command line arg.

**Comment by beansrowning** [comment]
@KarenKailun @scotthlee 

Leaving this as a PR for now, but this should be ready to go.

# New workflow  

From clone, you'll just run two scripts to pre-process data and run all models for every outcome:  

1. run_preprocessing.sh
2. run_all_models.sh  


# Changes

## Shell Scripts  
- run_all_models.sh is the new script for running all models across all outcomes and producing CIs
    - This could still use some error handling to halt the script on failure, but generally works.   
- run_models.sh now doesn't run CI script by default  
- run_models.sh also runs class-weighted fit for DAN and LSTM by default

## CI Script (python/analysis.py)
- CI script now checks the predictions spreadsheet for the model names, so there's no need for these to be pre-specified  
- new `--parallel` flag switches BCa computation to parallel execution (defaults false, but this is specified in run_all_models)  

## Baseline Models (python/baseline_models.py)  
- Out and data directories are now optional and do not have a default value
- All paths are constructed relative to script path, with command line arguments available to override if needed
- Subdirectories of the output folder are created if they don't exist at runtime to avoid errors when we go to write out stats
- All probs dumped into pickle dict with cutpoint used for downstream analysis

## Keras Models DAN/LSTM (python/model.py)
- Default batch size increased to 128, which apparently works now and fits much quicker
- Paths handled in same way as baseline models to avoid path construction issues
- weighted loss added as command line option to use class weights in loss computation
- TARGET -> OUTCOME to align with variable name in other scripts
- tf.data generator shennanigans resolved by setting epochs = 1, which doesn't really make a lot of sense but ¯\\_(ツ)_/¯  
- DAN and LSTM stats write-out streamlined

## Stats helpers (python/tools/analysis.py)  
- Whitespace changes, PEP8 compliance via yapf
- ROC-AUC error in multiclass case resolved by softmaxing logits from the sigmoid output of LSTM/DAN  
- Still 2 merge_cis() functions (@scotthlee which one do we want?)  
- imports streamlined based on usage

## Multiprocessing helpers (python/tools/multi.py)  
- boot_cis and jacknife_metrics now mirror non-Pool variants (but this should probably be handled as one entity to avoid divergence)  
- Streamlined imports based on usage
